lib/timezone/lookup/geonames.rb

```diff
@@ -34,13 +34,37 @@ def lookup(lat, long)
         raise(Timezone::Error::GeoNames, e.message)
       end
 
+      def lookup_with_etc(lat, long)
+        response = client.get(url(lat, long))
+
+        return unless response.body
+
+        data = JSON.parse(response.body)
+
+        return data['timezoneId'] if data['timezoneId']
+
+        return get_etc(data['gmtOffset']) if data['gmtOffset']
+
+        return unless data['status']
+
+        raise(Timezone::Error::GeoNames, data['status']['message'])
+      rescue => e
+        raise(Timezone::Error::GeoNames, e.message)
+      end
+
       private
 
+      def get_etc(offset)
+        return 'Etc/UTC' if offset == 0
+        "Etc/GMT#{format('%+d', offset.to_i)}"
```

Comment:
Yes, it makes sense.

Though, I don't think it's really guessing. Well, mostly not guessing. Guessing would be if gmtOffset depended on longitude only, but Geonames API returns gmtOffset with consideration of the real timezones borders. For example, let's take the the Pacific Ocean, for lat=20,long=165 it returns gmtOffset=12, but for lat=-20,long=165 it returns gmtOffset=11. Even in the ocean the borders of timezones are not straight, in this case we can see timezone change in a huge mostly empty area Marshall Islands.

On the other hand there are certain areas where offset is fractional, i.e. for the sea near Norfolk Island with lat=-29,long=167 returned gmtOffset is 11.5. And for such rare cases it would be really guessing as there are no Etc timezones with fractional offset. I just truncate such offset to the integer. In fact for Norfolk Island it would be the right thing to do since they changed their offset to 11 last October (Geonames just does not know about it obviously).

But anyway, I agree with making it optional for lookup method and putting it in the configuration.